### PR TITLE
documentation update re. JSON corpora

### PR DESCRIPTION
**Pull request by lukavdplas**
Many parts in the documentation were out of date because they assumed that a corpus is always based on a Python definition. Now that database-only corpora are becoming more-or-less functional, this is no longer accurate.

This inlcude some changes in how the documentation is structured, and some updates to docstrings and descriptions in the JSON schema.

This should bring the whole thing up to date with some PRs that are still open (namely #1569 , #1547 , #1567).

### CODE REVIEW

File: backend/addcorpus/python_corpora/corpus.py [hunk 1]
```diff
@@ -352,49 +354,68 @@ class FieldDefinition(Field):
...
-                 visualization_sort=None,
-                 es_mapping={'type': 'text'},
-                 language=None,
-                 search_filter=None,
-                 extractor=extract.Constant(None),
-                 sortable=None,
-                 searchable=None,
-                 downloadable=True,
-                 required=False,
+                 name: str = None,
```
Comment (JeltevanBoheemen) [inline backend/addcorpus/python_corpora/corpus.py:400]:
```suggestion
                 name: Optional[str] = None,
```

File: backend/addcorpus/python_corpora/corpus.py [hunk 2]
```diff
@@ -352,49 +354,68 @@ class FieldDefinition(Field):
...
+                 display_name: Optional[str] = None,
+                 display_type: Optional[str] = None,
+                 description: str = '',
+                 indexed: bool = True,
+                 hidden: bool = False,
+                 results_overview: bool = False,
+                 csv_core: bool = False,
+                 search_field_core: bool = False,
+                 visualizations: List[str] = [],
+                 visualization_sort: str = None,
```
Comment (JeltevanBoheemen) [inline backend/addcorpus/python_corpora/corpus.py:410]:
```suggestion
                 visualization_sort: Optional[str] = None,
```

**Review by JeltevanBoheemen — COMMENTED** [review]
Really nice!

File: backend/addcorpus/python_corpora/corpus.py (hunk 1 above)
Comment (lukavdplas) [reply backend/addcorpus/python_corpora/corpus.py:400]: I wasn't sure about this. `None` is not actually valid input here.

File: backend/addcorpus/python_corpora/corpus.py [hunk 3]
```diff
@@ -352,49 +354,68 @@ class FieldDefinition(Field):
...
+                 es_mapping: Dict = {'type': 'text'},
+                 language: Optional[str] = None,
+                 search_filter: Optional[Filter] = None,
+                 extractor: extract.Extractor = extract.Constant(None),
+                 sortable: Optional[bool] = None,
+                 searchable: Optional[bool] = None,
```
[the remark this answers is not in the document]
Comment (lukavdplas) [reply backend/addcorpus/python_corpora/corpus.py:416]: `None` means "let the constructor figure it out" - which sets these properties to `True` or `False` depending on other properties. `False` means you "force" `False` as the value.

**Pull request merged**